### DISCUSSION TITLE
RTH scenario

### DISCUSSION THREAD

**Discussion by samsab1** [iNavFlight/inav#10193] [Q&A]
I have just tested the RTH with speedybee f405v4 firmware 7.1.2 .

the precision of the RTH is incredible.
But I'm not sure I've mastered the scenario, as the options proposed by INAV are not easy to understand.
how can I ensure that the following steps take place in the event of RTH?
Step 1: climb or descend to set altitude (50 m) before any other action.
Step 2: return to the last track back point recorded on take-off
Step 3: return to the initial take-off trajectory.

who can explain to me, with pictures, how to set up inav?
thanks for your help

Translated with DeepL.com (free version)

**Comment by MrD-RC** [comment]
Have you checked out the Wiki? All RTH modes are explained with images for help. 

https://github.com/iNavFlight/inav/wiki/Navigation-Mode:-Return-to-Home

**Comment by samsab1** [comment]
Thanks Darren
didn't see this wiki before

**Comment by samsab1** [comment]
not so obvious Darren.
will it be possible for you to tell me how achieve my request?

Step 1: climb or descend to set altitude (50 m) before any other action.
Step 2: return to the last track back point recorded on take-off
Step 3: return to the initial take-off trajectory.

**Reply by sensei-hacker** [reply]
> not so obvious Darren. will it be possible for you to tell me how achieve my request?
> 
> Step 1: climb or descend to set altitude (50 m) before any other action. Step 2: return to the last track back point recorded on take-off Step 3: return to the initial take-off trajectory.

You can do 1 or 2, followed by 3. Option one is listed under "RTH Altitude control modes" and "Climb First".

Option 2 is covered under Trackback:

Trackback
...

When triggered the craft returns along the trackback route until the end is reached at which point it reverts to normal RTH heading directly home. _**It doesn't perform the RTH climb phase at the start of the Trackback**_ but instead uses the altitude recorded for each track point so long as that altitude is no lower than the altitude when Trackback was triggered (this is the minimum altitude used throughout the Trackback phase).

Landing is covered here:
https://github.com/iNavFlight/inav/blob/master/docs/Fixed%20Wing%20Landing.md

**Comment by samsab1** [comment]
thanks for the answer

i work on it........
by the way can you answser to that:

on the first takeoff, the initial trajectory is used to define the track back points.
question1: these points are defined with what spacing?
question 2: the initial trajectory is memorized over what maximum total distance?
question 3: do these memorized points also include the altitude, or are they just a projection on the ground?
question 4: can the initial climb plan after takeoff be exactly the descent plan in the context of an RTH?
